GVFS/GVFS.Common/Git/GitProcess.cs

```diff
@@ -699,20 +699,38 @@ public class Result
             public const int SuccessCode = 0;
             public const int GenericFailureCode = 1;
 
-            public Result(string output, string errors, int returnCode)
+            public Result(string stdout, string stderr, int exitCode)
             {
-                this.Output = output;
-                this.Errors = errors;
-                this.ReturnCode = returnCode;
+                this.Output = stdout;
+                this.Errors = stderr;
+                this.ExitCode = exitCode;
             }
 
             public string Output { get; }
             public string Errors { get; }
-            public int ReturnCode { get; }
+            public int ExitCode { get; }
 
-            public bool HasErrors
+            public bool ExitCodeIsSuccess
             {
-                get { return this.ReturnCode != SuccessCode; }
+                get { return this.ExitCode == Result.SuccessCode; }
+            }
+
+            public bool ExitCodeIsFailure
+            {
+                get { return !this.ExitCodeIsSuccess; }
+            }
+
+            public bool StderrContainsErrors()
+            {
+                if (!string.IsNullOrWhiteSpace(this.Errors))
+                {
+                    return !this.Errors
+                         .Trim()
+                         .Split('\n')

```

Comment:
Should we specify `StringSplitOptions.RemoveEmptyEntries` here? Otherwise a blank line between warning messages would cause us to return false here, right?